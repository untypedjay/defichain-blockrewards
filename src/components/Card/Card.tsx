import React from 'react';
import styled from 'styled-components';
import { cardStyles, StyledContent, StyledLabel } from '../../styles/CardStyle';

interface Props {
  title: string;
  children: string | JSX.Element;
  label?: string;
}

const StyledContainer = styled.div`
  width: 100%;
`;

const StyledCard = styled.div`
  ${cardStyles};
`;

const StyledHeading = styled.h2`
  font-size: 1.1rem;
`;

export default function Card({ title, children, label }: Props): JSX.Element {
  return (
    <StyledContainer>
      <StyledHeading>{ title }</StyledHeading>
      <StyledCard>
        {
          typeof children === 'string' ? <StyledContent>{ children }</StyledContent> :
          children
        }
        { label && <StyledLabel>{ label }</StyledLabel> }
      </StyledCard>
    </StyledContainer>
  );
}