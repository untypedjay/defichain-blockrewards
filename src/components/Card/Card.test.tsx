import ReactDOM from 'react-dom';
import React from 'react';
import Card from './Card';
import { cleanup, render } from '@testing-library/react';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card title="Title" label="Label">Test</Card>, div);
});

it('renders correctly', () => {
  render(<Card title="Title" label="Label">Test</Card>);
});

it('matches snapshot', () => {
  const { asFragment } = render(<Card title="Title" label="Label">Test</Card>);
  expect(asFragment()).toMatchSnapshot();
});