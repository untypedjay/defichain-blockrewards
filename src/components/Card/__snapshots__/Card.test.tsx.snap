// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-dlfnbm ioFdWY"
  >
    <h2
      class="sc-eCssSg eROQwk"
    >
      Title
    </h2>
    <div
      class="sc-hKgILt aqgCd"
    >
      <p
        class="sc-bdfBwQ dHtfKI"
      >
        Test
      </p>
      <p
        class="sc-gsTCUz fEJTeQ"
      >
        Label
      </p>
    </div>
  </div>
</DocumentFragment>
`;
